### PR TITLE
WT-13468 Split the cache statistics into individual group stastics

### PR DESCRIPTION
**Pull request by kommiharibabu**
Currently, plenty of cache statistics exist in the WiredTiger. Splitting them into individual statistics groups makes tracking and debugging an issue easy.

The planned additional groups are:

Eviction-server
Eviction-thread
Eviction-application-thread

### CODE REVIEW

File: dist/stat_data.py [hunk 1]
```diff
@@ -507,6 +457,80 @@ def __init__(self, name, desc, flags=''):
     DhandleStat('dh_sweep_tod', 'connection sweep time-of-death sets'),
     DhandleStat('dh_sweeps', 'connection sweeps'),
 
+    ##########################################
+    # Eviction server statistics
+    ##########################################
+    EvictionServerStat('cache_eviction_internal_pages_already_queued', 'internal pages seen by eviction walk that are already queued'),
```
Comment (quchenhao) [inline dist/stat_data.py:463]: Should this now starts with eviction instead of cache_eviction?

File: dist/stat_data.py [hunk 2]
```diff
@@ -267,75 +279,13 @@ def __init__(self, name, desc, flags=''):
...
-    CacheStat('cache_eviction_server_skip_trees_eviction_disabled', 'eviction server skips trees that disable eviction'),
-    CacheStat('cache_eviction_server_skip_trees_not_useful_before', 'eviction server skips trees that were not useful before'),
-    CacheStat('cache_eviction_server_skip_trees_stick_in_cache', 'eviction server skips trees that are configured to stick in cache'),
-    CacheStat('cache_eviction_server_skip_trees_too_many_active_walks', 'eviction server skips trees because there are too many active walks'),
-    CacheStat('cache_eviction_server_skip_unwanted_pages', 'eviction server skips pages that we do not want to evict'),
-    CacheStat('cache_eviction_server_skip_unwanted_tree', 'eviction server skips tree that we do not want to evict'),
-    CacheStat('cache_eviction_server_slept', 'eviction server slept, because we did not make progress with eviction'),
-    CacheStat('cache_eviction_slow', 'eviction server unable to reach eviction goal'),
     CacheStat('cache_eviction_stable_state_workers', 'eviction worker thread stable number', 'no_clear'),
```
Comment (quchenhao) [inline dist/stat_data.py:287]: Should these start with only cache instead of cache_eviction?

**Comment by ershov** [comment]
Please make sure that there are no derived stats or [Alexanraia](https://wiki.corp.mongodb.com/pages/viewpage.action?spaceKey=WT&title=Alexandria+FTDC+Analysis) rules based on the renamed ones.

File: dist/stat_data.py (hunk 2 above)
Comment (kommiharibabu) [reply dist/stat_data.py:287]: There are some stats that are common across eviction server, eviction threads and eviction is part of the cache. So I feel it is better to have them as `cache_eviction...`

File: dist/stat_data.py (hunk 1 above)
Comment (kommiharibabu) [reply dist/stat_data.py:463]: I agree with you about renaming them into `eviction` by removing the `cache` prefix. Currently, I want to limit the changes as part of this PR. I will create another ticket to handle the renaming of the stats.

**Review by quchenhao — APPROVED** [review]
lgtm

**Comment by kommiharibabu** [comment]
@ershov Thanks for the T2 link, yes, there are some stats expecting `cache` as a prefix by T2. I will create a ticket to update them.

**Comment by ershov** [comment]
@kommiharibabu I actually propose to not rename stats. It will create a lot of issues across different versions of WT and T2.

**Comment by kommiharibabu** [comment]
Thanks @ershov. After discussing with @Sean04, Currently, T2 doesn't handle WT version-specific statistics and it is difficult to maintain the stats when they are renamed. I will close this PR.

**Comment by ershov** [comment]
Should we roll back https://github.com/wiredtiger/wiredtiger/pull/11003 too?

**Comment by kommiharibabu** [comment]
No. That change was renaming a `stat name` and not its `description`. The diagnostic tools use the `description` and the `stat name` is used internally in WT. So there is no problem.

**Pull request closed without merging**